addon-test-support/@ember/test-helpers/setup-context.ts

```diff
@@ -59,34 +59,33 @@ registerWarnHandler((message, options, next) => {
   next.apply(null, [message, options]);
 });
 
-export interface BaseContext {
-  [key: string]: unknown;
-}
+export type BaseContext = object;
 
 /**
  * The public API for the test context, which test authors can depend on being
  * available.
  *
  * Note: this is *not* user-constructible; it becomes available by calling
- * `setupContext()` with a `BaseContext`.
+ * `setupContext()` with a base context object.
  */
 export interface TestContext extends BaseContext {
   owner: Owner;
 
   set<T>(key: string, value: T): T;
   setProperties<T extends Record<string, unknown>>(hash: T): T;
   get(key: string): unknown;
-  getProperties(...args: string[]): Pick<BaseContext, string>;
+  getProperties(...args: string[]): Record<string, unknown>;
 
   pauseTest(): Promise<void>;
   resumeTest(): void;
 }
 
 // eslint-disable-next-line require-jsdoc
 export function isTestContext(context: BaseContext): context is TestContext {
+  let maybeContext = context as Record<string, unknown>;
   return (
-    typeof context['pauseTest'] === 'function' &&
-    typeof context['resumeTest'] === 'function'
+    typeof maybeContext['pauseTest'] === 'function' &&
+    typeof maybeContext['resumeTest'] === 'function'
```

Comment:
For a possible follow-on (keeping type changes separate from behavioral changes):

```ts
return (
  'pauseTest' in context && 
  typeof maybeContext['pauseTest'] === 'function' &&
  'resumTest' in context && 
  typeof maybeContext['resumeTest'] === 'function'
);
```

Reply:
I had that and my editor was happy with it, but it turns out the `in` narrowing for `object` landed in TS 4.9, and definitely didn't want to touch that as part of this 😅